File: efishery-nodejs/modules/auth/controller.js
const { httpResponse } = require("../../utilities/response");
const Service = require("./service");

module.exports = class controller {
    constructor(option) {
        this.service = option?.service || new Service();
    }

    async login(req, res) {
        httpResponse(await this.service.login(req.body), res);
    }

    async register(req, res) {
        httpResponse(await this.service.register(req.body), res);
    }

    async checkToken(req, res) {
        const token = req.header("authorization").split(' ')[1] ?? '';
        httpResponse(await this.service.checkToken(token), res);
    }
}

File: efishery-golang/modules/auth/controller.go
package auth

import (
	"efishery-golang/dto"
	"github.com/gin-gonic/gin"
)

type Controller struct {
	service ServiceInterface
}

func NewController(service ServiceInterface) Controller {
	return Controller{
		service: service,
	}
}

func (ctrl Controller) Login(body LoginRequest) (res dto.BaseResponse, err error) {
	data, err := ctrl.service.Login(body)
	if err != nil {
		return dto.DefaultErrorBaseResponseWithMessage("Error"), nil
	}
	if data == "" {
		return dto.DefaultErrorBaseResponseWithMessage("Invalid phone or password"), nil
	}
	res = dto.BaseResponse{
		Success:      true,
		MessageTitle: "Success",
		Message:      "Success",
		Data: LoginResponse{
			Token: data,
		},
	}
	return res, nil
}

func (ctrl Controller) register(body RegisterRequest) (res dto.BaseResponse, err error) {
	data, err := ctrl.service.Register(body)
	if err != nil {
		return dto.DefaultErrorBaseResponseWithMessage("Error"), nil
	}
	if data.ID == 0 {
		return dto.DefaultErrorBaseResponseWithMessage("Phone already used"), nil
	}
	res = dto.BaseResponse{
		Success:      true,
		MessageTitle: "Success",
		Message:      "Success",
		Data:         data,
	}
	return res, nil
}

func (ctrl Controller) CheckToken(c *gin.Context) (res dto.BaseResponse, err error) {
	authDataValue, _ := c.Get("authData")
	res = dto.BaseResponse{
		Success:      true,
		MessageTitle: "Success",
		Message:      "Success",
		Data:         authDataValue,
	}
	return res, nil
}
